packages/network-controller/src/NetworkController.test.ts

```diff
@@ -3722,6 +3724,103 @@ describe('NetworkController', () => {
       });
     });
   });
+
+  describe('upsertNetworkConfiguration', () => {
+    it('add a network configuration', async () => {
+      const messenger = buildMessenger();
+      await withController({ messenger }, async ({ controller }) => {
+        const rpcUrlNetwork = {
+          chainId: undefined,
+          nickname: 'RPC',
+          rpcPrefs: undefined,
+          rpcUrl: 'rpc_url',
+          ticker: 'RPC',
+        };
+
+        expect(controller.state.networkConfigurations).toStrictEqual({});
+
+        controller.upsertNetworkConfiguration(rpcUrlNetwork);
+
+        expect(
+          Object.values(controller.state.networkConfigurations),
+        ).toStrictEqual(expect.arrayContaining([rpcUrlNetwork]));
+      });
+    });
+
+    it('update a network configuration when the configuration being added has an rpcURL that matches an existing configuration', async () => {
```

Comment:
Sorry, I'm an idiot. I just realized that the implementation accounts for multiple network configurations having the same RPC URL and updates all of them.

1) Should we keep doing this? I assume this code has been around for a while and is there to handle data that existed before we added a check that prevents multiple networks from existing that point to the same RPC URL.
2) Assuming we are keeping this logic (for now) then do you mind testing with multiple network configurations here? Again sorry I didn't notice that before 😬

Reply:
So at the moment (atleast on the extension) we don't allow users to configure multiple networks with the same RPC URL so where you say

 > updates all of them.
 
this would be referring to only one of the network. In the tests we could of course create an artificial scenario in state where multiple networks with the same RPC URL are configured preconfigured - in which case t[he for loop here](https://github.com/MetaMask/core/pull/1064/files#diff-ea0aa37793484e278bd59970b7a6ce8380b75b723b34612081d402fb66261cfdR492) would hit one of them and update it and leave the other one unchanged = but that wouldn't be testing existing UX.

Now @Gudahtt has mentioned that in the future we may want to change the requirements to that `rpcUrl` is not necessarily unique. This would be related to permissioned RPC providers where we would add additional properties to the `NetworkConfiguration` object with API keys or some credentials sent via request headers.

Reply:
Gotcha. I guess we can hold off until that change is made.